File: C++/3.单例模式/SingletonPatternDemo.cpp
/**
 * @file SingletonPatternDemo.cpp
 * @author {Layne} ({shu_huanglei@163.com})
 * @brief
 * @version 0.1
 * @date 2019-10-16
 *
 * @copyright Copyright (c) 2019
 *
 */
#include "SingletonHunger.h"
#include "SingletonLazy.h"

int main(int argc, char const *argv[]) {
    SingletonHunger &hunger1 = SingletonHunger::getInstance();
    hunger1.ShowMessage();
    SingletonHunger &hunger2 = SingletonHunger::getInstance();
    hunger2.ShowMessage();

    SingletonLazy &lazy1 = SingletonLazy::getInstance();
    lazy1.ShowMessage();
    SingletonLazy &lazy2 = SingletonLazy::getInstance();
    lazy2.ShowMessage();

    return 0;
}

File: Objective-C/3.SingletonPattern/SingletonPatternDemo.m
//
//  SingletonPatternDemo.m
//  3.SingletonPattern
//
//  Created by Layne on 2020/10/18.
//

#import "SingleObject.h"

int main(int argc, const char* argv[]) {
    @autoreleasepool {
        //获取唯一可用的对象
        SingleObject* object = [SingleObject getInstance];

        //显示消息
        [object showMessage];
    }
    return 0;
}
